### ISSUE TITLE
Parse error with question mark followed by dot in ternary operator 

### ISSUE THREAD

**Issue by ocean90** [mck89/peast#48]
Peast currently fails to parse this code:

```php
$source = "var a=b?.05:-.05";
$ast = Peast\Peast::latest($source)->parse();
```

```
PHP Fatal error:  Uncaught Peast\Syntax\Exception: Unexpected: 05 in /private/tmp/peast/vendor/mck89/peast/lib/Peast/Syntax/ParserAbstract.php:305
Stack trace:
#0 /private/tmp/peast/vendor/mck89/peast/lib/Peast/Syntax/Parser.php(3423): Peast\Syntax\ParserAbstract->error()
#1 /private/tmp/peast/vendor/mck89/peast/lib/Peast/Syntax/Parser.php(3269): Peast\Syntax\Parser->parseLeftHandSideExpression()
#2 /private/tmp/peast/vendor/mck89/peast/lib/Peast/Syntax/Parser.php(3220): Peast\Syntax\Parser->parsePostfixExpression()
#3 /private/tmp/peast/vendor/mck89/peast/lib/Peast/Syntax/Parser.php(3131): Peast\Syntax\Parser->parseUnaryExpression()
#4 /private/tmp/peast/vendor/mck89/peast/lib/Peast/Syntax/Parser.php(3093): Peast\Syntax\Parser->parseLogicalBinaryExpression()
#5 /private/tmp/peast/vendor/mck89/peast/lib/Peast/Syntax/Parser.php(3049): Peast\Syntax\Parser->parseConditionalExpression()
#6 /private/tmp/peast/vendor/mck89/peast/lib/Peast/Syntax/Parser.php(2911): Peast\Syntax\Parser->parseAssignmentExpression()
#7 /private/tmp/peast in /private/tmp/peast/vendor/mck89/peast/lib/Peast/Syntax/ParserAbstract.php on line 305
```

<details>
<summary>AST based on https://astexplorer.net/</summary>

```json
{
  "type": "Program",
  "start": 0,
  "end": 16,
  "body": [
    {
      "type": "VariableDeclaration",
      "start": 0,
      "end": 16,
      "declarations": [
        {
          "type": "VariableDeclarator",
          "start": 4,
          "end": 16,
          "id": {
            "type": "Identifier",
            "start": 4,
            "end": 5,
            "name": "a"
          },
          "init": {
            "type": "ConditionalExpression",
            "start": 6,
            "end": 16,
            "test": {
              "type": "Identifier",
              "start": 6,
              "end": 7,
              "name": "b"
            },
            "consequent": {
              "type": "Literal",
              "start": 8,
              "end": 11,
              "value": 0.05,
              "raw": ".05"
            },
            "alternate": {
              "type": "UnaryExpression",
              "start": 12,
              "end": 16,
              "operator": "-",
              "prefix": true,
              "argument": {
                "type": "Literal",
                "start": 13,
                "end": 16,
                "value": 0.05,
                "raw": ".05"
              }
            }
          }
        }
      ],
      "kind": "var"
    }
  ],
  "sourceType": "module"
}
```

</details>

Same issue for `{a:b?.05:-.05}`. But there's no parse error when the dot is not followed by 0 as in `var a=b?.5:-.05`. `var a=b?0.05:-.05` is working too.

Initially reported in https://github.com/wp-cli/i18n-command/issues/297.

**Comment by mck89** [comment]
The number 0 was not handled correctly after the "?." sequence, i've fixed it, thank you for reporting!

**Comment by ocean90** [comment]
@mck89 Thank you! Confirming that the fix is working for me.

**Comment by mck89** [comment]
Released version 1.13.11
